File: webf/lib/src/geometry/dom_matrix.dart
/*
 * Copyright (C) 2024-present The OpenWebF Company. All rights reserved.
 * Licensed under GNU GPL with Enterprise exception.
 */
/*
 * Copyright (C) 2022-2024 The WebF authors. All rights reserved.
 */


import 'dom_matrix_readonly.dart';

class DOMMatrix extends DOMMatrixReadOnly {
  DOMMatrix(super.context, super.domMatrixInit);

  DOMMatrix.fromMatrix4(super.context, super.matrix4, super.flag2D) : super.fromMatrix4();
}

File: bridge/core/platform/geometry/dom_matrix.cc
/*
 * Copyright (C) 2022-present The WebF authors. All rights reserved.
 */

#include "dom_matrix.h"

namespace webf {

DOMMatrix* DOMMatrix::Create(ExecutingContext* context,
                             const std::shared_ptr<QJSUnionDomStringSequenceDouble>& init,
                             ExceptionState& exception_state) {
  return MakeGarbageCollected<DOMMatrix>(context, init, exception_state);
}

DOMMatrix::DOMMatrix(ExecutingContext* context,
                     const std::shared_ptr<QJSUnionDomStringSequenceDouble>& init,
                     ExceptionState& exception_state)
    : DOMMatrixReadonly(context, init, exception_state) {}

}  // namespace webf